unittests/test_pipeline.py

```diff
@@ -343,21 +344,35 @@ def test_sourcepath_upref(self):
     @rt.switch_runtime(fixtures.TEST_SITE_CONFIG, 'testsys')
     def test_extra_resources(self):
         # Load test site configuration
-        test = rfm.RegressionTest()
+        test = rfm.RunOnlyRegressionTest()
         test._prefix = 'unittests/resources/checks'
         test.valid_prog_environs = ['*']
         test.valid_systems = ['*']
-        test.extra_resources = {
-            'gpu': {'num_gpus_per_node': 2},
-            'datawarp': {'capacity': '100GB', 'stagein_src': '/foo'}
-        }
+        test.executable = 'echo'
+        test.executable_opts = ['Hello World!']
+        test.sanity_patterns = sn.assert_found(r'Hello World!', test.stdout)
+        test.local = True
         partition = rt.runtime().system.partition('gpu')
         environ = partition.environment('builtin-gcc')
         test.setup(partition, environ)
+
+        # set extra resources and job options here to make sure that they are
+        # put in the job script after setup
+        test.extra_resources = {
+            'gpu': {'num_gpus_per_node': 2},
+            'datawarp': {'capacity': '100GB', 'stagein_src': test.stagedir}
+        }
         test.job.options += ['--foo']
+
+        test.run()
+        test.wait()
+        test.check_sanity()
+        test.check_performance()
+        test.cleanup(remove_files=True)
```

Comment:
You should use the `_run()` function defined in this test file, instead of running yourself the test like that.

Reply:
I want to set the `extra_resources` between `setup` and `run` while `run` does both.

Reply:
Do it properly with the `run_after` decorator. Actually if you define the test the right way. See a bit above how this is done.